presto-main/src/main/java/com/facebook/presto/operator/TableWriterOperator.java

```diff
@@ -164,6 +184,7 @@ public void addInput(Page page)
     {
         requireNonNull(page, "page is null");
         checkState(state == State.RUNNING, "Operator is %s", state);
+        checkState(blocked == NOT_BLOCKED, "output is already blocked");
```

Comment:
Oh, sorry I see what you are doing... I'd just call `checkState(needsInput, "Operator does not need input")`, and this will handle both conditions